components/core/DataView.js

```diff
@@ -240,26 +212,54 @@ export default class DataView extends React.Component {
       window.addEventListener("remote-slate-object-remove", this._handleRemoteSlateObjectRemove);
       window.addEventListener("remote-slate-object-add", this._handleRemoteSlateObjectAdd);
     }
+
+    window.addEventListener("scroll", this._handleScroll);
+    await this._handleCheckScroll();
   }
 
   componentWillUnmount() {
     mounted = false;
     window.removeEventListener("remote-data-deletion", this._handleDataDeletion);
     window.removeEventListener("remote-slate-object-remove", this._handleRemoteSlateObjectRemove);
     window.removeEventListener("remote-slate-object-add", this._handleRemoteSlateObjectAdd);
+    window.removeEventListener("scroll", this._handleCheckScroll);
+    window.removeEventListener("remote-update-carousel", this._handleUpdate);
   }
 
-  _increment = (direction) => {
-    if (
-      direction > 0 &&
-      this.state.startIndex + VIEW_LIMIT < this.props.viewer.library[0].children.length
-    ) {
-      this.setState({ startIndex: this.state.startIndex + VIEW_LIMIT });
-    } else if (direction < 0 && this.state.startIndex - VIEW_LIMIT >= 0) {
-      this.setState({ startIndex: this.state.startIndex - VIEW_LIMIT });
+  _handleDebounce = (func, wait) => {
+    let timeout;
```

Comment:
this reference `timeout` will get lost since its in the scope of this closure, you need to add this as a class member

```
class Component extends React.Component {
  _timeout = null;

  _handleDebounce = (callbackFn, waitMs) => {
    return function executedFunction(...args) {
      const debouncedCall = () => {
        callbackFn(...args);
      }

      clearTimeout(this._timeout);
      this._timeout = setTimeout(debouncedCall, waitMs)
    }
  }
}
```